docs/quickstart/dart.md

```diff
@@ -0,0 +1,231 @@
+---
+bodyclass: docs
+layout: docs
+headline: Dart Quickstart
+sidenav: doc-side-quickstart-nav.html
+type: markdown
+---
+<p class="lead">This guide gets you started with gRPC in Dart with a simple
+working example.</p>
+
+<div id="toc"></div>
+
+## Before you begin
+
+### Prerequisites
+
+#### Dart SDK
+
+gRPC requires Dart SDK version 1.24.3 or higher.
+
+For installation instructions, follow this guide: [Install Dart](https://www.dartlang.org/install)
+
+#### Install Protocol Buffers v3
+
+While not mandatory to use gRPC, gRPC applications usually leverage Protocol
+Buffers v3 for service definitions and data serialization, and our example code
+uses Protocol Buffers as well as gRPC.
+
+The simplest way to install the protoc compiler is to download pre-compiled
+binaries for your platform (`protoc-<version>-<platform>.zip`) from here:
+[https://github.com/google/protobuf/releases](https://github.com/google/protobuf/releases)
+
+  * Unzip this file.
+  * Update the environment variable `PATH` to include the path to the protoc
+    binary file.
+
+Next, install the protoc plugin for Dart
+
+```sh
+$ pub global activate protoc_plugin
+```
+
+The compiler plugin, `protoc-gen-dart`, will be installed in `$HOME/.pub-cache/bin`.
+It must be in your $PATH for the protocol compiler, protoc, to find it.
+
+```sh
+$ export PATH=$PATH:$HOME/.pub-cache/bin
+```
+
+## Download the example
+
+You'll need a local copy of the example code to work through this quickstart.
+Download the example code from our Github repository (the following command
+clones the entire repository, but you just need the examples for this quickstart
+and other tutorials):
+
+```sh
+$ # Clone the repository at the latest release to get the example code:
+$ git clone https://github.com/grpc/grpc-dart
+$ # Navigate to the "Hello World" Dart example:
+$ cd grpc-dart/example/helloworld
+```
+
+## Try it!
+
+From the `example/helloworld` directory:
+
+1. Download package dependencies
+
+   ```sh
+   $ pub get
+   ```
+
+2. Run the server
+
+   ```sh
+   $ dart bin/server.dart
+   ```
+
+3. In another terminal, run the client
+
+   ```sh
+   $ dart bin/client.dart
+   ```
+
+Congratulations! You've just run a client-server application with gRPC.
+
+## Update a gRPC service
+
+Now let's look at how to update the application with an extra method on the
+server for the client to call. Our gRPC service is defined using protocol
+buffers; you can find out lots more about how to define a service in a `.proto`
+file in [gRPC Basics: Dart][]. For now all you need to know is that both the
+server and the client "stub" have a `SayHello` RPC method that takes a
+`HelloRequest` parameter from the client and returns a `HelloReply` from the
+server, and that this method is defined like this:
+
+
+```
```

Comment:
I think generally we are missing `dart` after the back-ticks in the start of the code blocks?

Reply:
+1 - just checked and it's supported by both the Github and Jekyll syntax highlighters.

Reply:
Dart code should already have `dart`. This is protobuf schema.

Reply:
That's fine.